File: simpsons/lib/models/episode.dart
class Episode {
  final String id;
  final int number;
  final String code;
  final String title;
  final String originalTitle;
  final String description;
  final String date_usa;

  Episode({
    required this.id,
    required this.number,
    required this.code,
    required this.title,
    required this.originalTitle,
    required this.description,
    required this.date_usa,
  });

  factory Episode.fromJson(Map<String, dynamic> json) {
    return Episode(
      id: json['_id'] as String,
      number: json['number'] as int? ?? 0,
      code: json['code'] as String? ?? 'Code inconnu',
      title: json['title'] as String? ?? 'Titre inconnu',
      originalTitle: json['original_title'] as String? ?? 'Titre original inconnu',
      description: json['description'] as String? ?? 'Description inconnue',
      date_usa: json['date_usa'] as String? ?? 'Date inconnue',
    );
  }
}

File: simpsons/backend/src/models/Episode.ts
import { Schema, model, Types } from 'mongoose';

interface Episode {
    _id?: Types.ObjectId;
    season: number;
    number: number;
    title: string;
    original_title?: string;
    description?: string;
    date_usa: Date;
}

const EpisodeSchema = new Schema<Episode>({
    _id: { type: Schema.Types.ObjectId, auto: true },
    season: { type: Number, required: true },
    number: { type: Number, required: true },
    title: { type: String, required: true },
    original_title: { type: String, required: false },
    description: { type: String, required: false },
    date_usa: { type: Date, required: true }
});

export default model<Episode>('Episode', EpisodeSchema);